### DISCUSSION TITLE
transport calculation (solve pressure drop and flow rate) for a viscous + inertial flow

### DISCUSSION THREAD

**Discussion by MTJKoning** [PMEAL/OpenPNM#2876] [Q&A]
Hey, 

I am currently using OpenPNM for my graduation project, where i want to simulate a packed bed reactor using a transient reaction/transport model. I have looked into the software and found that there only exists a transport model using stokes flow (pure viscous flow, linear). I want to validate the model over a range of flow rates such that they resemble the Ergun equation (viscous and inertial flow, non-linear). I was wondering if this is even possible using OpenPNM, and if so where should I begin? Or am i better off using a linear stokes flow, or solving the non-linearity outside of OpenPNM?

Kind regards,

Menno Koning

**Comment by jgostick** [comment]
We don't support high-inertia flows and TBH I have not really considered how to do it so not sure how tricky it would be.  If you can find a solid literature source that describe it, including the numerical implementation, then send it to me or post it here and I'll take a look to see how it could be done.

**Reply by MTJKoning** [reply]
I was afraid that would be the case... Thanks for your quick reply though!

I am currently thinking of using this paper to guide me through it;

https://doi.org/10.1016/j.ces.2023.118626

It looks quite similar, exept that it solves a non-linear flow and pressure field. It uses an iterative method with direct substitution to solve the system of equations. I was thinking: OpenPNM algorithm ReactiveTransport uses a non-linear newton iterative solver to solve for the source terms (linearizing them). Would it not be possible to use this as well for the flow field (which would have the form dP= A*q + B*q^2)?

I do have some difficulty in finding which solver it actually uses, since all solvers in the solver section are linear solvers. Furthermore I have trouble finding the sore spot of where to alter what, if i were to use the nonlinear solver of ReactiveTransport.

I hope you understand my problem and could help me on the way a bit. I will try using the OpenPNM solver first, and if that does not work I will simply try to solve the flow field outside of the PNM using an iterative Newton solver, and set the flow field as input for the reactive model.

**Reply by MTJKoning** [reply]
https://doi.org/10.1016/j.powtec.2023.119103
